### PR TITLE
Printf: Add support for tabs, and give helpful error messages

### PR DESCRIPTION
**Pull request by mwachs5**
**Related issue**: 
https://github.com/freechipsproject/chisel3/issues/1321

I added support for \t in printf. I also made the `require` statements give more information if you do something wrong which is how I figured out the tab was the problem.

**Type of change**: bug report

**Impact**: API modification

**Development Phase**: implementation

**Release Notes**

Printables printf now supports \t characters.

### CODE REVIEW

File: chiselFrontend/src/main/scala/chisel3/Printf.scala [hunk 1]
```diff
@@ -19,13 +19,15 @@ object printf { // scalastyle:ignore object.name
...
+      require(x.toInt >= 0, s"char ${x} to Int ${x.toInt} must be >= 0")
       if (x == '"' || x == '\\') {
         s"\\${x}"
       } else if (x == '\n') {
         "\\n"
-      } else {
-        require(x.toInt >= 32) // TODO \xNN once FIRRTL issue #59 is resolved
+      } else if (x == '\t') {
+        "\\t"
+      }else {
```
Comment (jackkoenig) [inline chiselFrontend/src/main/scala/chisel3/Printf.scala]:
```suggestion
      } else {
```

**Comment by jackkoenig** [comment]
Oh I didn't realize this was against `3.2.x`, we usually PR against master and then backport. That's okay, I'll forward port this.

**Comment by mwachs5** [comment]
oh, are the tests passing now? cuz it failed before...

**Comment by jackkoenig** [comment]
Yeah, you fixed the failing test

**Pull request merged**